pkg/deviceshifu/deviceshifusocket/deviceshifusocket_test.go

```diff
@@ -126,3 +134,169 @@ func TestEncodeMessage(t *testing.T) {
 		t.Errorf("not match with current output, output: %v", output)
 	}
 }
+
+func TestCollectSocketTelemetry(t *testing.T) {
+	socketProtocol := v1alpha1.ProtocolSocket
+	httpProtocol := v1alpha1.ProtocolHTTP
+	address := "localhost:3000"
+	emptyAddress := ""
+	db := &deviceshifubase.DeviceShifuBase{
+		Name: "Unit Test",
+		EdgeDevice: &v1alpha1.EdgeDevice{
+			Spec: v1alpha1.EdgeDeviceSpec{
+				Protocol: &socketProtocol,
+				Address:  &address,
+			},
+		},
+	}
+	ds := &DeviceShifu{
+		base: db,
+	}
+
+	listener, err := net.Listen("tcp", "localhost:3000")
+	if err != nil {
+		t.Errorf("Cannot Listen at port 3000")
+	}
+
+	go func() {
+		_, _ = listener.Accept()
+	}()
+	// testcase pass
```

Comment:
I suggest we adopt table driven unit test if we can.
please refer to https://dev.to/boncheff/table-driven-unit-tests-in-go-407b

The benefit of doing this is:
1. structured unit test
2. easy to maintain unit test

in some case if the test case is pretty simple to manage, we can ignore this practice